### PR TITLE
[WICKED] Startandstop test 4

### PR DESCRIPTION
**Pull request by jlausuch**
Bridge - ifup, remove one config, ifreload

    When I create a bridge on interface eth1 from legacy files
    Then I delete the config file for br1
    #  
    When I bring up all by ifreload
    Then there should not be the br1 card anymore
    But the dummy1 card should still be there
    And the eth1 card should still be there

- Related ticket: https://progress.opensuse.org/issues/41660
- Verification run: http://fromm.arch.suse.de/tests/2026

### CODE REVIEW

**Comment by jlausuch** [comment]
@asmorodskyi  @cfconrad

**Comment by okurz** [comment]
LGTM

File: tests/wicked/startandstop_sut.pm [hunk 1]
```diff
@@ -79,9 +79,20 @@ sub run {
     assert_script_run("rm /etc/sysconfig/network/ifcfg-$iface $config $dummy");
     assert_script_run("wicked ifreload --timeout infinite all");
     my $res = script_run("ip link|grep 'dummy0\|br0'");
-    $results{3} = $res ? "PASSED" : "FAILED";
+    $results{3} = $res ne 0 ? "PASSED" : "FAILED";
```
[the remark this answers is not in the document]
Comment (jlausuch) [reply tests/wicked/startandstop_sut.pm]: true and fixed :)

File: tests/wicked/startandstop_sut.pm [hunk 2]
```diff
@@ -79,9 +79,20 @@ sub run {
     assert_script_run("rm /etc/sysconfig/network/ifcfg-$iface $config $dummy");
     assert_script_run("wicked ifreload --timeout infinite all");
     my $res = script_run("ip link|grep 'dummy0\|br0'");
-    $results{3} = $res ? "PASSED" : "FAILED";
+    $results{3} = $res != 0 ? "PASSED" : "FAILED";
```
Comment (asmorodskyi) [inline tests/wicked/startandstop_sut.pm]:
```suggestion
    $results{3} = $res ? "FAILED" : "PASSED";
```

File: tests/wicked/startandstop_sut.pm [hunk 3]
```diff
@@ -79,9 +79,20 @@ sub run {
...
 
+    $self->before_scenario('Test 4', 'Bridge - ifup, remove all config, ifreload', $iface);
+    $self->get_from_data('wicked/ifcfg/br0',    $config);
+    $self->get_from_data('wicked/ifcfg/dummy0', $dummy);
+    $self->setup_bridge($config, $dummy, 'ifup');
+    assert_script_run("rm $config");
+    assert_script_run("wicked ifreload --timeout infinite all");
+    my $res1 = script_run("ip link|grep br0");
+    my $res2 = script_run("ip link|grep dummy0");
+    $results{4} = $res1 != 0 && $res2 == 0 ? "PASSED" : "FAILED";
```
Comment (asmorodskyi) [inline tests/wicked/startandstop_sut.pm]:
```suggestion
    $results{4} = $res1 && ! $res2 ? "PASSED" : "FAILED";
```

File: tests/wicked/startandstop_sut.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (asmorodskyi) [reply tests/wicked/startandstop_sut.pm]: it is even simpler no need to do the job which ternary operator  is actually suppose to do

File: tests/wicked/startandstop_sut.pm (hunk 3 above)
Comment (jlausuch) [reply tests/wicked/startandstop_sut.pm]: ok, but actually 
    
    $results{4} = $res1 && ! $res2 ? "FAILED" : "PASSED";

File: tests/wicked/startandstop_sut.pm (hunk 2 above)
Comment (jlausuch) [reply tests/wicked/startandstop_sut.pm]: this should be

    $results{3} = $res ? "PASSED" : "FAILED";

File: tests/wicked/startandstop_sut.pm (hunk 3 above)
Comment (jlausuch) [reply tests/wicked/startandstop_sut.pm]: ok

**Pull request merged**